scripts/apps/search/components/HighlightsInfo.jsx

```diff
@@ -28,6 +31,24 @@ export class HighlightsInfo extends React.Component {
 
         closeActionsMenu();
         this.renderDropdown();
+        this.setActionMenuState(true);
+    }
+
+    componentWillUnmount() {
+        if (this.state.isActionMenuOpen) {
```

Comment:
Instead of adding in this state, why not just change `scripts/apps/search/helpers.js`? It could be:
```js
export function closeActionsMenu() {
    let el = menuHolderElem();

    if (el) {
        ReactDOM.unmountComponentAtNode(el);
    }
}
```
Then you can remove all this checking and state setting.

Reply:
Just checking if the menu element exists won't work. It will close the action menu of any other item. Hence the state is set for that particular item.

Reply:
I think it's true that there is only one menu at a time, so if we set somewhere what item has the menu opened it should be enough instead of having the state all over the place

Reply:
A quick solution might be to add an ID to the open menu element, and optionally pass it to `closeActionMenu`. If it's provided, then only close the menu with that ID (if it's open).

Reply:
I will make change accordingly. @petrjasek, @gbbr